### PR TITLE
First draft of running halfstack in ci

### PR DESCRIPTION
**Pull request by internet-diglett**
This PR is for adding some jobs that will build and run a falcon topology to test the latest build of the netstack image.

There's some additional cleanup I can do in this PR once the following PRs are merged:
- [x] https://github.com/oxidecomputer/testbed/pull/6
- [x] https://github.com/oxidecomputer/falcon/pull/16
- [x] #2 
- [x] https://github.com/oxidecomputer/testbed/pull/7

### CODE REVIEW

File: .github/buildomat/jobs/netstack-prepare.sh [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#: name = "netstack-prepare"
+#: variety = "basic"
+#: target = "helios"
+#: rust_toolchain = "nightly"
+#:
+#: output_rules = [
+#:   "=/out/build-tools.tar.xz",
+#: ]
+#:
+#: access_repos = [
```
Comment (rcgoodfellow) [inline .github/buildomat/jobs/netstack-prepare.sh:14]: I think the only access repo that's needed here is testbed. All the other stuff is public.

File: .github/buildomat/jobs/netstack-prepare.sh [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+export RUST_LOG=debug
+
+# List the host's interfaces
+dladm show-link
+topdir=$(pwd)
+
+mkdir bin
+
+pushd bin
+curl -OL https://github.com/stedolan/jq/releases/download/jq-1.4/jq-solaris11-64
```
Comment (rcgoodfellow) [inline .github/buildomat/jobs/netstack-prepare.sh:43]: is there something wrong with the `jq` from `pkg`?

File: .github/buildomat/jobs/netstack-validation.sh [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+# This script is used to validate the netstack image
+# It builds a falcon topology using the latest netstack image,
+# then tests communication across the simulated network fabric.
+#:
+#: name = "netstack-validation"
+#: variety = "basic"
+#: target = "lab"
+#: rust_toolchain = "nightly"
+#:
+#: access_repos = [
```
Comment (rcgoodfellow) [inline .github/buildomat/jobs/netstack-validation.sh:11]: same comment on access repos, only testbed needed here.

File: .github/buildomat/jobs/netstack-prepare.sh (hunk 2 above)
Comment (internet-diglett) [reply .github/buildomat/jobs/netstack-prepare.sh:43]: `jq` doesn't exist on the baremetal ci server (it's _very_ barebones), so we have to bring our own. I think I tried copying the one that was installed but it was dynamically linked to a library that the baremetal server didn't have.

File: .github/buildomat/jobs/netstack-prepare.sh (hunk 1 above)
Comment (internet-diglett) [reply .github/buildomat/jobs/netstack-prepare.sh:14]: Removed

File: .github/buildomat/jobs/netstack-validation.sh (hunk 3 above)
Comment (internet-diglett) [reply .github/buildomat/jobs/netstack-validation.sh:11]: Removed

**Review by rcgoodfellow — APPROVED** [review]
Awesome to see this working. LGTM 🚀

**Pull request merged**